tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/PcmUmlclassViewFactory.xtend

```diff
@@ -0,0 +1,88 @@
+package tools.vitruv.applications.pcmumlclass.tests
+
+import tools.vitruv.testutils.TestViewFactory
+import tools.vitruv.framework.views.View
+import org.eclipse.uml2.uml.Model
+import org.eclipse.xtend.lib.annotations.FinalFieldsConstructor
+import org.palladiosimulator.pcm.repository.Repository
+import org.palladiosimulator.pcm.repository.RepositoryComponent
+import org.palladiosimulator.pcm.system.System
+
+@FinalFieldsConstructor
+class PcmUmlclassViewFactory extends TestViewFactory {
+	
+	private def View createUmlView() {
+		createViewOfElements("UML", #{Model})
+	}
+
+	private def View createPcmView() {
+		createViewOfElements("Pcm packages and components", #{Repository, RepositoryComponent})
+	}
+
+	private def View createUmlAndPcmClassesView() {
+		createViewOfElements("UML and Pcm components", #{RepositoryComponent, Model})
+	}
+
+	private def View createUmlAndPcmPackagesView() {
+		createViewOfElements("UML and Pcm packages", #{Repository, Model})
+	}
+	
+	private def View createUmlAndPcmSystemView() {
+		createViewOfElements("UML and Pcm packages", #{System, Model})
+	}
+
+	/**
+	 * Changes the UML view containing all UML models as root elements 
+	 * according to the given modification function. 
+	 * Records the performed changes, commits the recorded changes, and closes the view afterwards.
+	 */
+	def void changeUmlView((View)=>void modelModification) {
+		changeViewRecordingChanges(createUmlView, modelModification)
+	}
+
+	/**
+	 * Changes the Java view containing all Java packages and classes as root elements 
+	 * according to the given modification function. 
+	 * Records the performed changes, commits the recorded changes, and closes the view afterwards.
+	 */
```

Comment:
Check the comments (everywhere in this file)

Reply:
Corrected in e644396f32b0509741a91d19d43f764532a6229a